### PR TITLE
Login signup feature

### PR DESCRIPTION
**Pull request by HuzaifaHussnain**
This branch contains the Login, Signup, logout and forget password functionality along with the Token authentication.

### CODE REVIEW

File: sound_cloud_backend/manage.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 """Django's command-line utility for administrative tasks."""
 import os
 import sys
-
+from dotenv import load_dotenv
```
Comment (hannankamrantintash) [inline sound_cloud_backend/manage.py:5]: Its fine to use dotenv for environment variables. Another option is python-decouple; if you go with the latter option, you don't have to write these two lines neither in manage.py nor in wsgi.py.

File: sound_cloud_backend/sound_cloud_backend/settings.py [hunk 2]
```diff
@@ -37,6 +37,10 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+
+    'rest_framework',
+    # local apps
```
Comment (hannankamrantintash) [inline sound_cloud_backend/sound_cloud_backend/settings.py]: Avoid using the comments in between the lines.

File: sound_cloud_backend/sound_cloud_backend/settings.py [hunk 3]
```diff
@@ -118,3 +122,15 @@
 # https://docs.djangoproject.com/en/3.0/howto/static-files/
 
 STATIC_URL = '/static/'
+
+# Settigs for S3 storage
+
+AWS_ACCESS_KEY_ID = os.environ.get('AWS_ACCESS_KEY_ID')
+AWS_SECRET_ACCESS_KEY = os.environ.get('AWS_SECRET_ACCESS_KEY')
+AWS_STORAGE_BUCKET_NAME = os.environ.get('AWS_STORAGE_BUCKET_NAME')
+
```
Comment (hannankamrantintash) [inline sound_cloud_backend/sound_cloud_backend/settings.py]: Follow the same practice for an extra line throughout the project.

File: sound_cloud_backend/user/api/serializers.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+from rest_framework import serializers
+from django.contrib.auth.models import User
+
+class UserSerializer(serializers.ModelSerializer):
+	class Meta:
+		model = User
+		fields = [
```
Comment (hannankamrantintash) [inline sound_cloud_backend/user/api/serializers.py]: You can specify all the fields in a single line.

File: sound_cloud_backend/user/api/views.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+from django.contrib.auth.models import User
+from .serializers import UserSerializer
+
+
+class UserCreateAPIView(generics.CreateAPIView):
+	queryset = User.objects.all()
+	serializer_class = UserSerializer
+
+	# def create(self, request, *args, **kwargs):
```
Comment (hannankamrantintash) [inline sound_cloud_backend/user/api/views.py]: Avoid from committing a dead code.

File: sound_cloud_backend/user/api/serializers.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+			'first_name',
+			'last_name',
+			'email',
+			'password'
+		]
+
+		extra_kwargs = {'password': {'write_only': True}}
+
+	def create(self, validated_data):
+		password = validated_data.pop('password')
```
Comment (hannankamrantintash) [inline sound_cloud_backend/user/api/serializers.py]: Make it a normal practice to add a docstring for each class and function. Currently, I think you haven't specified it anywhere.

File: sound_cloud_backend/sound_cloud_backend/restconf/main.py [hunk 7]
```diff
@@ -1,6 +1,6 @@
 REST_FRAMEWORK = {
     'DEFAULT_AUTHENTICATION_CLASSES': [
-        'rest_framework_simplejwt.authentication',
```
Comment (hannankamrantintash) [inline sound_cloud_backend/sound_cloud_backend/restconf/main.py]: Any specific reason why you moved from jwt authentication to django token authentication?

Comment (HuzaifaHussnain) [reply sound_cloud_backend/sound_cloud_backend/restconf/main.py]: I can't remember why i replaced jwt with TokenAuthentication.

**Pull request merged**